### DISCUSSION TITLE
Understanding Starlette WebSocket and TCP Flow Control

### DISCUSSION THREAD

**Discussion by viett-appcard** [encode/starlette#2704] [Q&A] [answered]
Hi,

I am implementing a websocket server with FastAPI and ran into a thought of what happens when the client doesn't read or is slow to read but doesn't disconnect?

I've read up briefly on tcp flow control and tried to find documentation for starlette's websocket but didn't find anything. Wondering if anyone can clear up or go into more detail.

**Comment by nggit** [comment, marked as answer]
You don't have to worry about that.
I haven't heard of ASGI apps getting involved with _flow control_ because that's the server's job to manage it.

You can assume that if things are not as expected then the bug is on the server. E.g. Uvicorn.
